python/pyspark/ml/tests.py

```diff
@@ -455,6 +455,14 @@ def test_logistic_regression_check_thresholds(self):
             LogisticRegression, threshold=0.42, thresholds=[0.5, 0.5]
         )
 
+    def test_preserve_set_state(self):
+        model = Binarizer()
+        self.assertFalse(model.isSet("threshold"))
+        model._transfer_params_to_java()
```

Comment:
yeah, it would be a little better to call the actual `transform`, but we would still need to call `_transfer_params_from_java` or check `isSet` with a direct call to Java via py4j.  I was going to do this, but the `ParamTest` class doesn't already create a `SparkSession` - I'm sure it's just a small amount of overhead but that's why I thought to just use `_transfer_params_to_java`.

Do you think it would be worth it to change `ParamTests` to inherit from `SparkSessionTestCase` so a session is created and I could make a `DataFrame` to transform?